Packages/nadena.dev.modular-avatar/Runtime/ModularAvatarObjectRemover.cs

```diff
@@ -0,0 +1,52 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+namespace nadena.dev.modular_avatar.core
+{
+    [ExecuteInEditMode]
+    [DisallowMultipleComponent]
+    [AddComponentMenu("Modular Avatar/MA Object Remover")]
+    public class ModularAvatarObjectRemover : AvatarTagComponent
+    {
+        public bool hideInHierarchy;
+
+        public ModularAvatarObjectRemover()
+        {
+            EditorApplication.delayCall += Refresh;
+        }
+
+        private void OnEnable()
+        {
+            Refresh();
+        }
+
+        private void Refresh()
+        {
+            if (Application.isPlaying) return;
+            gameObject.SetActive(false);
+            if (hideInHierarchy) Hide();
+            // components can be always restored from prefabs, so removing them now does not break anything and will
+            // provide better compatibility with other tools
+            foreach (var componentsInChild in gameObject.GetComponentsInChildren<MeshRenderer>())
+            {
+                if (PrefabUtility.IsPartOfAnyPrefab(componentsInChild)) DestroyImmediate(componentsInChild);
+            }
+            foreach (var componentsInChild in gameObject.GetComponentsInChildren<SkinnedMeshRenderer>())
+            {
+                if (PrefabUtility.IsPartOfAnyPrefab(componentsInChild)) DestroyImmediate(componentsInChild);
+            }
+        }
+
+        public void Hide()
+        {
+            hideInHierarchy = true;
+            gameObject.hideFlags |= HideFlags.HideInHierarchy;
```

Comment:
When you make changes to a prefab instance, it's important to call `PrefabUtility.RecordPrefabInstancePropertyModifications` (and `EditorUtility.SetDirty`, in case it's not part of a prefab instance) to ensure the changes are saved properly.

Reply:
This changes cannot be saved to a prefab anyways: https://forum.unity.com/threads/prefab-sub-asset-hideflags-dont-persist.1324920/